graylog2-server/src/main/java/org/graylog2/security/realm/HTTPHeaderAuthenticationRealm.java

```diff
@@ -65,24 +66,23 @@ public HTTPHeaderAuthenticationRealm(ClusterConfigService clusterConfigService,
         this.authServiceAuthenticator = authServiceAuthenticator;
         this.trustedProxies = trustedProxies;
 
-        setAuthenticationTokenClass(HttpHeadersToken.class);
+        setAuthenticationTokenClass(RemoteAddressAuthenticationToken.class);
         setCachingEnabled(false);
         // Credentials will be matched via the authentication service itself so we don't need Shiro to do it
         setCredentialsMatcher(new AllowAllCredentialsMatcher());
     }
 
     @Override
     protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token) throws AuthenticationException {
-        final HttpHeadersToken headersToken = (HttpHeadersToken) token;
+        final RemoteAddressAuthenticationToken remoteAddrToken = (RemoteAddressAuthenticationToken) token;
         final HTTPHeaderAuthConfig config = loadConfig();
 
         if (!config.enabled()) {
             LOG.debug("Skipping disabled HTTP header authentication");
             return null;
         }
 
-        final MultivaluedMap<String, String> headers = headersToken.getHeaders();
-        final Optional<String> optionalUsername = headerValue(headers, config.usernameHeader());
```

Comment:
The method `#headerValue` can now be removed as it's unused after this change.

Reply:
:white_check_mark: